Scripting-Engine/Scripting-Engine/LeagueSandbox-Default/Champions/Ezreal/Q.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Numerics;
+using LeagueSandbox.GameServer.Logic.GameObjects;
+using LeagueSandbox.GameServer.Logic.API;
+
+namespace Ezreal
+{
+    class Q
+    {
+        static void onStartCasting(Champion owner)
+        {
+            ApiFunctionManager.AddParticleTarget(owner, "ezreal_bow.troy", owner, 1, "L_HAND");
+
+        }
+        static void onFinishCasting(Champion owner, Spell spell)
+        {
+            Vector2 current = new Vector2(owner.X, owner.Y);
+            Vector2 to = Vector2.Normalize(new Vector2(spell.X, spell.Y) - current);
+            Vector2 range = to * 1150;
+            Vector2 trueCoords = current + range;
+
+            spell.AddProjectile("EzrealMysticShotMissile", trueCoords.X, trueCoords.Y);
```

Comment:
Projectiles do need to be linked to spells in some scenarios AFAIK though, however, I'm uncertain if I'm right about this, so no need to listen to me here.